### DISCUSSION TITLE
The workbench failed to connect to the server (Error: Time limit reached)

### DISCUSSION THREAD

**Discussion by ptyork** [community/community#46084] [Codespaces]
### Select Topic Area

Bug

### Body

Suddenly I'm seeing this for all codespaces in all repositories in my organization. Basically the codespace server seems to start okay. The VS Code workbench starts and then hangs until a pop-up displays the message in the title.

Opening the same codespace in VS Code gives me "Could not fetch remote environment" and "Failed to connect to the remote extension host server (Error: WebSocket close with status code 1006)" messages. Opening developer tools gives me a more detailed exception for the WebSocket timeout which I can share privately if it is helpful.

Is Codespaces down? Is it just MY organization? How can I debug / fix this?

**Comment by MrBam44** [comment]
It's possible that there is an issue with the GitHub Codespaces service, but it's also possible that it could be an issue with your network or your codespace configuration. To check if it's a service-wide issue, you can check the GitHub status page: https://www.githubstatus.com/

If it's not a service-wide issue, you can try the following steps to debug and fix the issue:

Check your network connection: Make sure you have a stable internet connection and try connecting to GitHub Codespaces again.

Restart the Codespace: Sometimes simply restarting the Codespace can resolve the issue.

Check the Codespace configuration: Ensure that the codespace is properly configured and that all required services are running.

Try opening the Codespace in a different browser: If the issue persists, try opening the Codespace in a different browser to see if it resolves the issue.

If you're still experiencing the issue, you can reach out to GitHub support for further assistance.

**Reply by ptyork** [reply]
Thanks! I actually thought this was my only support option. I dug deeper into the help pages and was able to find where you can create actual support tickets ("Contact Us" to the rescue). I had actually already checked all of the bits you suggested (at least to the best of my knowledge), so I've submitted a ticket. When it's resolved, I'll respond back here so that others who encounter similar symptoms can benefit from the solution.

**Reply by dlastrazeneca** [reply]
@ptyork : any solution?

**Reply by ptyork** [reply]
> @ptyork : any solution?

@dlastrazeneca not generically. The issue here was eventually tracked down to a change made by GitHub that interacted badly with the use of a custom user name and a specific plugin. https://github.com/orgs/community/discussions/46232

I never responded here since it seemed pretty obscure and they never actually fixed it for my existing spaces. Students just lost access to them. :(
